packages/nimbus/src/components/date-range-picker/date-range-picker.dev.mdx

```diff
@@ -0,0 +1,407 @@
+
+### Date value types
+
+Three primary date types are available depending on your needs:
+
+```tsx
+import { CalendarDate, CalendarDateTime, ZonedDateTime } from '@internationalized/date';
+
+// Date only (no time)
+const dateOnly = new CalendarDate(2024, 1, 15);
+
+// Date with time (no timezone)
+const dateTime = new CalendarDateTime(2024, 1, 15, 14, 30);
+
+// Date with time and timezone
+const zonedDate = new ZonedDateTime(2024, 1, 15, 14, 30, 'America/New_York');
+```
+
+### Creating date values
+
+Use helper functions to parse ISO strings or create dates programmatically:
+
+```tsx
+import { parseDate, parseDateTime, parseZonedDateTime } from '@internationalized/date';
+
+// Parse ISO date strings
+const date = parseDate('2024-01-15');
+const dateTime = parseDateTime('2024-01-15T14:30');
+const zonedDate = parseZonedDateTime('2024-01-15T14:30[America/New_York]');
+```
+
+### Date range structure
+
+The component expects and returns date ranges in this structure:
+
+```tsx
+type DateRange = {
+  start: CalendarDate | CalendarDateTime | ZonedDateTime;
+  end: CalendarDate | CalendarDateTime | ZonedDateTime;
+}
+```
+
+> [!TIP]\
+> See [Adobe's @internationalized/date documentation](https://react-spectrum.adobe.com/internationalized/date/index.html) for complete API reference and advanced usage.
+
+## Usage examples
+
+### Size options
+
+Two size variants are available to match your interface density:
+
+```jsx-live-dev
+const App = () => (
+  <Stack direction="column" gap="400">
+    <DateRangePicker size="sm" />
+    <DateRangePicker size="md" />
+  </Stack>
+)
+```
+
+Use `size="sm"` for compact layouts or dense data tables. Use `size="md"` (default) for standard forms and prominent date selection.
+
+### Visual variants
+
+Choose between solid and ghost variants to match your design context:
+
+```jsx-live-dev
+const App = () => (
+  <Stack direction="column" gap="400">
+    <DateRangePicker variant="solid" />
+    <DateRangePicker variant="ghost" />
+  </Stack>
+)
+```
+
+The `solid` variant (default) provides clear visual boundaries. The `ghost` variant offers a minimal appearance suitable for embedded contexts.
+
+### Granularity options
+
+The `granularity` prop controls both the level of date precision and the component's behavior:
+
+```jsx-live-dev
+const App = () => (
+  <Stack direction="column" gap="400">
+    <DateRangePicker granularity="day" />
+    <DateRangePicker granularity="minute" />
+  </Stack>
+)
+```
+
+**Behavioral differences:**
+- `granularity="day"` (default): Calendar closes automatically after selecting the end date. Time inputs are hidden.
+- `granularity="minute"` or other time values: Calendar remains open after date selection, displaying time input fields in the footer. This allows users to set both dates and times in one interaction.
+
+Available granularity options: `"day"`, `"hour"`, `"minute"`, `"second"`.
+
+### Controlled state
```

Comment:
We should also consider having an `uncontrolled` section that shows how to get the selected value from the component